onnxruntime/core/framework/kernel_registry_manager.h

```diff
@@ -42,17 +42,15 @@ class KernelRegistryManager {
   // Then B > A > providers
   void RegisterKernelRegistry(std::shared_ptr<KernelRegistry> kernel_registry);
 
-  // This function assumes the node is already assigned to an execution provider
-  // Don't call this function before graph partition is done
-  Status CreateKernel(const onnxruntime::Node& node, const IExecutionProvider& execution_provider,
-                      const SessionState& session_state,
```

Comment:
Don't need a 'search+create' variant of CreateKernel. SearchKernelRegistry is used to find the KernelCreateInfo, and the new CreateKernel takes that as input. This removes duplication of the search logic.